### PR TITLE
OBD2 Updates codes

### PR DESCRIPTION
**Pull request by imhoffj**
Updates codes and documentation per feedback. Making a pull request again r/2.8.8 as requested.

### CODE REVIEW

**Comment by stieg** [comment]
ok to test

File: src/OBD2/OBD2.c [hunk 1]
```diff
@@ -138,9 +382,9 @@ int OBD2_request_PID(unsigned char pid, int *value, size_t timeout)
     msg.data[7] = 0x55;
     msg.dataLength = 8;
     msg.isExtendedAddress = 0;
-
+
```
Comment (stieg) [inline src/OBD2/OBD2.c]: Extra whitespace.

**Comment by stieg** [comment]
Huh... I'm wondering why my CODING_STANDARD.md and CONTRIBUTING.md docs are showing up in your patch.  Mayhaps I didn't merge them into r/2.8.8.  Anyway nuke those few white spaces in the patch and I will merge it in.  Thanks!

File: src/OBD2/OBD2.c (hunk 1 above)
Comment (imhoffj) [inline src/OBD2/OBD2.c]: Updated to remove whitespace

File: src/OBD2/OBD2.c [hunk 2]
```diff
@@ -56,69 +56,313 @@ static float kph_to_mph(float kph)
...
-
+    
+    if (msg->addressValue == STANDARD_PID_RESPONSE &&
+        msg->data[0] >= 3 &&
+        msg->data[1] == CUSTOM_MODE_SHOW_CURRENT_DATA &&
+        msg->data[2] == pid ) {
+        
         result = 1;
-
+
```
Comment (stieg) [inline src/OBD2/OBD2.c]: Missed white space.  Applies throughout.

**Comment by stieg** [comment]
Missed some other stragglers at the top.

**Comment by imhoffj** [comment]
Should be good - what IDE are you using? I usually use XCode and not catching for some reason. Let me know if you see any others - thanks.

**Comment by stieg** [comment]
I use emacs.  I have an on-save hook that murders white space.  Looks like you got them all.  In this goes.  Thanks.

**Pull request merged**